### PR TITLE
Automated spellcheck via GitHub Actions (and addressing all surfaced issues)

### PR DESCRIPTION
**Pull request by SeanKilleen**
## Describe this PR

* Adds automated spell-checking via a GitHub Action
* Addresses all the issues that cSpell surfaces

(Background: I've recently done this for NUnit, Marten, Playwright, Uno Platform, etc. and figured you might also find it beneficial. No worries about rejecting it if you're not interested since I didn't raise an issue first.)

## Changes

* Adds automated spell-checking via a GitHub Action
* Addresses all the issues that cSpell surfaces

## What issue does this fix?

Any open spell-check issues hopefully. 😄 But honestly in a brief glance I didn't see any that applied.

### CODE REVIEW

**Comment by SeanKilleen** [comment]
cSpell found 176 errors to address (these are usually 98% existing words that need to be added or things that back-ticks solve. I'll work to address those next.

**Comment by SeanKilleen** [comment]
Alright, this is ready for review I think. I'll do a quick self review to point out information.

File: cSpell.json
Comment (SeanKilleen) [inline cSpell.json:4]: ℹ️ Quick note: `words` here will pop up as spelling suggestions in IDEs; `ignoreWords` will not.

File: content/300-guides/050-database/100-developing-with-prisma-migrate/165-enable-native-database-functions.mdx
Comment (SeanKilleen) [inline content/300-guides/050-database/100-developing-with-prisma-migrate/165-enable-native-database-functions.mdx:3]: ℹ️ Wanted to call your attention to this because I changed the Front Matter property name, assuming the thing referencing it uses the correct name. So code may need to change elsewhere to allow fixing this spelling issue.

File: content/600-about/200-prisma-docs/20-style-guide/01-writing-style.mdx
Comment (SeanKilleen) [inline content/600-about/200-prisma-docs/20-style-guide/01-writing-style.mdx:33]: ℹ️ I put these behind back-ticks because otherwise `behaviour` would show as a spelling error. But I didn't want to add it to the dictionary because it's a specific case we want to catch. Putting the terms in back-ticks allows them to be there without a spelling issue being raised and without compromising readability IMO.

File: content/200-concepts/050-overview/100-what-is-prisma/100-data-modeling.mdx
Comment (janpio) [inline content/200-concepts/050-overview/100-what-is-prisma/100-data-modeling.mdx]: I think more
```suggestion
Depending on the domain of your application, the models will be different. For example, if you're writing a blogging application, you might have models such as _blog_, _author_, _article_. When writing a car sharing app, you probably have models like _driver_, _car_, _route_. Application models enable you to represent these different entities in your code by creating respective _data structures_.
```

File: content/200-concepts/100-components/02-prisma-client/061-custom-validation.mdx
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/061-custom-validation.mdx]: Hmmm https://en.wiktionary.org/wiki/upvote 🤔

File: content/300-guides/050-database/100-developing-with-prisma-migrate/200-troubleshooting-development.mdx
Comment (janpio) [inline content/300-guides/050-database/100-developing-with-prisma-migrate/200-troubleshooting-development.mdx]: Hmmm https://en.wiktionary.org/wiki/hotfix 🤔

File: content/300-guides/200-deployment/110-deployment-guides/300-deploying-to-vercel.mdx
Comment (janpio) [inline content/300-guides/200-deployment/110-deployment-guides/300-deploying-to-vercel.mdx]: Hmm https://en.wiktionary.org/wiki/preconfigure 🤔

File: content/400-reference/300-database-reference/01-database-features.mdx
Comment (janpio) [inline content/400-reference/300-database-reference/01-database-features.mdx]: Hmm https://en.wiktionary.org/wiki/autoincrement 🤔

**Review by janpio — COMMENTED** [review]
Some nice fixes, and some things where we will need to discuss what we want from a style perspective.

**Comment by SeanKilleen** [comment]
@janpio no worries -- this PR isn't about being "correct". Any change that a maintainer is comfortable with codifying is fine to me. All the ones you pulled out seemed reasonable and I'll make the appropriate changes to revert them and add them to the dictionary. 👍

File: content/200-concepts/050-overview/100-what-is-prisma/100-data-modeling.mdx
Comment (SeanKilleen) [reply content/200-concepts/050-overview/100-what-is-prisma/100-data-modeling.mdx]: D'oh! Nice catch, thanks.

File: content/200-concepts/100-components/02-prisma-client/061-custom-validation.mdx
Comment (SeanKilleen) [reply content/200-concepts/100-components/02-prisma-client/061-custom-validation.mdx]: ✅ Reverted

File: content/300-guides/200-deployment/110-deployment-guides/300-deploying-to-vercel.mdx
Comment (SeanKilleen) [reply content/300-guides/200-deployment/110-deployment-guides/300-deploying-to-vercel.mdx]: ✅ Reverted

File: content/400-reference/300-database-reference/01-database-features.mdx
Comment (SeanKilleen) [reply content/400-reference/300-database-reference/01-database-features.mdx]: ✅ Reverted

File: content/300-guides/050-database/100-developing-with-prisma-migrate/200-troubleshooting-development.mdx
Comment (SeanKilleen) [reply content/300-guides/050-database/100-developing-with-prisma-migrate/200-troubleshooting-development.mdx]: ✅ Reverted.

**Comment by SeanKilleen** [comment]
@janpio for your consideration -- I'd love to get this merged soon given that conflicts are starting to creep in. I've addressed all your comments (which were totally reasonable) and I think this is ready for review / merge.

**Comment by janpio** [comment]
Would love to, but I am on the coding side here - @tanberry and @keerlu are actually on the docs side and can take a look.

**Comment by tanberry** [comment]
Hi hi @SeanKilleen wow, let me take a closer look, but I'll first say Thank you very much. ;-)

**Comment by tanberry** [comment]
Yeah for sure this tool caught a lot of typos.. I think I recognize one of mine, errrrr....

A couple of questions, @SeanKilleen . In the cSpell.json file, the list of words starting at line 5.. are these words the tool is looking for, are these "approved spelling" words, what exactly?

I agree with the vast majority of the changes, though not these ones: 
- "car sharing" needs a hyphen
- "e-commerce" no longer needs a hyphen

I'd like to get our WebDev lead to take a look and make sure she agrees about adding the two new files, and the process. We have a pretty complex set of tools already, so I want to check with her and make sure this isn't overloading tings. But it sure seems to work well, so thank you again and we will get back to you soon.

**Comment by SeanKilleen** [comment]
@tanberry thanks for the review!

> I think I recognize one of mine, errrrr....

Better docs with a typo here and there than no docs. 😆 

> In the cSpell.json file, the list of words starting at line 5.. are these words the tool is looking for, are these "approved spelling" words, what exactly?

Sorry, I believe I had mentioned this in a self-review that somehow got resolved. `words` are accepted spellings -- they could be suggested by a spellchecker in the IDE (e.g. if someone is working within VS Code, the spell checker uses the cSpell dictionary). `ignoreWords` are words that will be not be suggested.

> "car sharing" needs a hyphen

That's fine. I suggested that originally I think but then reverted based on feedback. I will make the change.

> "e-commerce" no longer needs a hyphen

Also fine with me. The goal of this PR is not to be correct but rather to codify whatever the maintainers would like to codify. 

> I'd like to get our WebDev lead to take a look and make sure she agrees about adding the two new files, and the process. We have a pretty complex set of tools already,

I'll make those changes while we wait for your colleague to review. 👍 It should be fairly straightforward, and I've implemented it for a number of repos (NUnit, Marten, Akka .NET, etc.) without hearing of any issues. Worst case, disabling the GitHub action would be all you need to do in order to prevent an issue.

**Comment by SeanKilleen** [comment]
@tanberry FYI the requested changes were made. As far as I can tell this one's out of my hands for now, but feel free to ping me with any questions or updates you'd like.

**Review by tanberry — APPROVED** [review]
This is great, thanks so very much @SeanKilleen !

**Pull request merged**
